### ISSUE TITLE
stage2: when there is an analysis error, the compiler still generates (incorrect) code and returns 0.

### ISSUE THREAD

**Issue by g-w1** [ziglang/zig#6976]
The self-hosted compiler exits 0 even when there is an error with the compilation:

```
[]$ ./zig-cache/bin/zig build-exe test.zig
test.zig:3:5: error: use of undeclared identifier 'c'
[]$ echo $?
0
```

**Comment by g-w1** [comment]
This issue has much bigger scope than I originally thought: 
![](https://i.imgur.com/devHxMI.png)
When a file has wrong syntax, it still builds and runs the file. I do not think this is the right behavior. Should I make this a separate issue?

**Comment by alexnask** [comment]
The linker's openPath function is called before the module is analyzed

**Comment by g-w1** [comment]
Ah. Thanks, this saves me a lot of debugging time.

**Comment by g-w1** [comment]
This seems like a pretty big issue? Here is the information I have gathered:
* The binary output file is always created regardless of whether the analysis was a success. This could be desired, I guess, to always create the elf for incremental compilation? I think that if the analysis fails, then there should be no file emitted/the file should not be modified.
* Nothing happens when analysis failing in performAllTheWork.
* This seems like a structural issue of what alexnask said that openPath is called before the module is analyzed.
I was going to try to tackle this (I tried, but didn't want to change the code of stage2 too much), but now that it is actually a beast in hiding, Ill leave it to someone who has a better idea where the compiler should go.
